fake-koji/src/main/java/org/fakekoji/jobmanager/OTool.java

```diff
@@ -0,0 +1,39 @@
+package org.fakekoji.jobmanager;
+
+import org.fakekoji.jobmanager.model.JobUpdateResults;
+import org.fakekoji.jobmanager.model.Project;
+import org.fakekoji.storage.StorageException;
+
+import java.util.List;
+
+public class OTool {
+
+    private final JobUpdater jobUpdater;
+
+    public OTool(final JobUpdater jobUpdater) {
+        this.jobUpdater = jobUpdater;
+    }
+
+    /**
+     * Regenerate all jobs. If job is mssing, is (re)created.
+     *
+     * @throws StorageException
+     */
+    public <T extends Project> JobUpdateResults regenerateAll(
+            String projectId,
+            Manager<T> projectManager
+    ) throws StorageException, ManagementException {
+        JobUpdateResults sum = new JobUpdateResults();
+        JenkinsJobUpdater.wakeUpJenkins();
+        final List<T> projects = projectManager.readAll();
+        for (final Project project : projects) {
+            if (projectId == null || project.getId().equals(projectId)) {
+                JobUpdateResults r = jobUpdater.regenerate(project);
+                sum = sum.add(r);
```

Comment:
hm, maybe the comment was rightr, but hten whole class is misleading

Reply:
Didn't know where to put this method

Reply:
you will need to find out

Reply:
The move of that method is unclear. Why new class? Why named as it is? 
If new class, then with correct name. Even two classes.